include/z64schedule.h

```diff
@@ -0,0 +1,248 @@
+#ifndef Z64SCHEDULE_H
+#define Z64SCHEDULE_H
+
+#include "ultra64.h"
+
+/*
+  Schedule is a subsystem that acts as a way to make decisions based on the
+  time and scene (and a limited selection of items). It is utilized by writing
+  a script that is encoded into bytecode and ran, returning the result in a
+  struct. The returned result can be a value or a encoded time value.
+
+  The scripts contain 2 kinds of instructions:
+  - Checks with branches (relative offsets, either 1-byte offsets (short, *_S),
+                          or 2-byte offsets (long, *_L))
+  - Returns
+
+  Scripts are stored as u8[]. They are built using the macros are the bottom of
+  this file. The scheduledis.py script can be used to convert any scripts in
+  actor data into the macros.
+*/
+
+// Macro to convert the time format used in the save struct into the format used in Schedule
+#define SCHEDULE_CONVERT_TIME(time) ((time) - 0x10000 / 360 * 90)
+#define SCHEDULE_TIME_NOW SCHEDULE_CONVERT_TIME(gSaveContext.time)
+
+typedef enum {
+    /* 00 */ SCHEDULE_CMD_ID_CHECK_FLAG_S,        // Checks if a weekEventReg flag is set and branches if so, short range branch
+    /* 01 */ SCHEDULE_CMD_ID_CHECK_FLAG_L,        // Checks if a weekEventReg flag is set and branches if so, long range branch
+    /* 02 */ SCHEDULE_CMD_ID_CHECK_TIME_RANGE_S,  // Checks if the current time is within the range of the two provided times and branches if so, short range branch
+    /* 03 */ SCHEDULE_CMD_ID_CHECK_TIME_RANGE_L,  // Checks if the current time is within the range of the two provided times and branches if so, long range branch
+    /* 04 */ SCHEDULE_CMD_ID_RET_VAL_L,           // Ends script and returns 2-byte value (Note: bugged as the return value size is only 1 byte in the struct)
+    /* 05 */ SCHEDULE_CMD_ID_RET_NONE,            // Ends script without returning anything
+    /* 06 */ SCHEDULE_CMD_ID_RET_EMPTY,           // Ends script and indicates return without changing existing value
+    /* 07 */ SCHEDULE_CMD_ID_NOP,                 // No-Op
+    /* 08 */ SCHEDULE_CMD_ID_CHECK_MISC_S,        // Special check based on items or masks and branches if check passes, short range branch
+    /* 09 */ SCHEDULE_CMD_ID_RET_VAL_S,           // Ends script and returns byte value
+    /* 10 */ SCHEDULE_CMD_ID_CHECK_SCENE_S,       // Checks if the current scene is SceneNum and branches if not, short range branch
+    /* 11 */ SCHEDULE_CMD_ID_CHECK_SCENE_L,       // Checks if the current scene is SceneNum and branches if not, long range branch
```

Comment:
```suggestion
    /* 10 */ SCHEDULE_CMD_ID_CHECK_SCENE_S,       // Checks if the current scene is not SceneNum and branches if so, short range branch
    /* 11 */ SCHEDULE_CMD_ID_CHECK_SCENE_L,       // Checks if the current scene is not SceneNum and branches if so, long range branch
```
I just realized this only branches if you're not in the scene. I think having the negative earlier in the sentence would be clearer.

What would you think about `SCHEDULE_CMD_ID_CHECK_NOT_IN_SCENE` to be clearer at the risk of verbosity?

Reply:
I hadn't noticed the flipped logic between this and the other checks. I went with `SCHEDULE_CMD_ID_CHECK_NOT_IN_SCENE` (and `SCHEDULE_CMD_ID_CHECK_NOT_IN_DAY`, which was the same way)